tests/integration/internal/logsgenerator/logsgenerator.go

```diff
@@ -0,0 +1,133 @@
+package logsgenerator
+
+import (
+	"fmt"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+/// This package wraps the logsgenerator utility in sumologic-kubernetes-tools in a procedural API
+/// easily consumable by tests.
+
+const deploymentSleepTime = 60 * 60 * 24 // how much time we spend sleeping after generating logs in a Deployment
+
+type LogsGeneratorOptions struct {
+	// For all of these options, 0 and "" respectively are treated as "not set"
+
+	// how long we should run in total
+	// the way Duration and TotalLogs interact, is that the program exits on the first condition achieved
+	// in particular, there isn't a way to make it continue running after generating a fixed number of logs
+	Duration uint
+	// how many log lines to generate in total
+	TotalLogs uint
+	// maximum number of logs generated per second
+	LogsThroughput uint
+	// maximum number of bytes generater per second
+	BytesThroughput uint
+	// should we print diagnostic messages?
+	Verbose bool
+
+	// Pattern options
+	// See https://github.com/SumoLogic/sumologic-kubernetes-tools/tree/main/src/rust/logs-generator#patterns
+
+	// These options control random pattern generation
+	RandomPatterns      uint
+	MinPatternLength    uint
+	MaxPatternLength    uint
+	KnownWordsInRatio   uint
+	RandomWordsInRatio  uint
+	RandomDigitsInRatio uint
+
+	// This option allows the pattern to be controlled directly
+	Pattern string
+}
+
+func NewDefaultGeneratorOptions() *LogsGeneratorOptions {
+	return &LogsGeneratorOptions{
+		Duration:            0,
+		TotalLogs:           0,
+		LogsThroughput:      0,
+		BytesThroughput:     0,
+		Verbose:             false,
+		RandomPatterns:      10,
+		MinPatternLength:    5,
+		MaxPatternLength:    20,
+		KnownWordsInRatio:   7,
+		RandomWordsInRatio:  2,
+		RandomDigitsInRatio: 1,
+		Pattern:             "",
+	}
+}
+
+func GetLogsGeneratorDeployment(
+	namespace string,
+	name string,
+	image string,
+	options *LogsGeneratorOptions,
```

Comment:
Any reason to use a pointer?

Reply:
I don't think so, I think I was working on Rust code in tools at the same time and got semantics mixed up. Will switch to a value.